addons/web/static/src/js/views/calendar/calendar_model.js

```diff
@@ -651,6 +671,13 @@ return AbstractModel.extend({
                         f.label = to_read[f.avatar_model][f.value];
                     });
                 }
+                if (filter.color_model && filter.field_color) {
+                    _.each(filter.filters, function (f) {
+                        if (!f.color_index) {
+                            f.color_index = color_filter[filter.color_model] && color_filter[filter.color_model][f.value];
```

Comment:
I would do that earlier. I might be wrong but it seems there is no need to define an object color_filter.